documentation/src/main/asciidoc/userguide/chapters/caching/Caching.adoc

```diff
@@ -1052,3 +1052,37 @@ hibernate.session_factory_name_is_jndi = false
+It's important to understand that trying to set up second level cache in this way is generally not a good idea for the following reasons:
+
+* The purpose of a JPA/Hibernate second level cache is to store entities/collections recently retrieved from database or to maintain results of recent queries.
+So, part of the aim of the second level cache is to have data accessible locally rather than having to go to the database to retrieve it everytime this is needed.
+Hence, if you decide to set the second level cache to be remote as well, you're losing one of the key advantages of the second level cache: the fact that the cache is local to the code that requires it.
+
+* Setting a remote second level cache can have a negative impact in the overall performance of your application because it means that cache misses require accessing a remote location to verify whether a particular entity/collection/query is cached.
+With a local second level cache however, these misses are resolved locally and so they are much faster to execute than with a remote second level cache.
+
+There are however some edge cases where it might make sense to have a remote second level cache, for example:
+
+* You are having memory issues in the JVM where JPA/Hibernate code and the second level cache is running.
+Off loading the second level cache to remote Hot Rod servers could be an interesting way to separate systems and allow you find the culprit of the memory issues more easily.
+
+* Your application layer cannot be clustered but you still want to run multiple application layer nodes.
+In this case, you can't have multiple local second level cache instances running because they won't be able to invalidate each other for example when data in the second level cache is updated.
+In this case, having a remote second level cache could be a way out to make sure your second level cache is always in a consistent state, will all nodes in the application layer pointing to it.
+
+* Rather than having the second level cache in a remote server, you want to simply keep the cache in a separate VM still within the same machine.
+In this case you would still have the additional overhead of talking across to another JVM, but it wouldn't have the latency of across a network.
+The benefit of doing this is that:
+
+** Size the cache separate from the application, since the cache and the application server have very different memory profiles.
+One has lots of short lived objects, and the other could have lots of long lived objects.
+
+**  To pin the cache and the application server onto different CPU cores (using _numactl_ ), and even pin them to different physically memory based on the NUMA nodes.
+
+If interested to try out using remote Infinispan second-level cache, the easiest way to do is via the <<caching-provider-jcache,JCache region factory>>.
```

Comment:
This section could also mention an alternative of using remote cache store (I've seen that used in production according one of the JIRAs), though I find this the least safe and non-tested way. At least for the replicated/dist caches. In such configuration, there's a problem that all regions (which create their own caches according to the templates) store the data in single remote cache, because 2LC can't multiplex the cache store part of configuration. That requires at least use of `default` cache keys factory.